cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaListenersReconciler.java

```diff
@@ -403,7 +404,8 @@ protected Future<Void> clusterIPServicesReady()   {
             String bootstrapAddress = getInternalServiceHostname(reconciliation.namespace(), ListenersUtils.backwardsCompatibleBootstrapServiceName(reconciliation.name(), listener), useServiceDnsDomain);
 
             if (listener.isTls()) {
-                result.bootstrapDnsNames.add(bootstrapAddress);
+                ModelUtils.generateAllServiceDnsNames(reconciliation.namespace(), ListenersUtils.backwardsCompatibleBootstrapServiceName(reconciliation.name(), listener))
```

Comment:
I think if this list bootstrapDnsNames is used only with certs, we can avoid to fill it here when ClusterCa.generateBrokerCerts already generate the same list and add dns names to subject. We do it twice now, here https://github.com/scholzj/strimzi-kafka-operator/blob/use-all-DNS-names-for-cluster-ip-listener-bootstrap-servive/cluster-operator/src/main/java/io/strimzi/operator/cluster/model/ClusterCa.java#L185 and here https://github.com/scholzj/strimzi-kafka-operator/blob/use-all-DNS-names-for-cluster-ip-listener-bootstrap-servive/cluster-operator/src/main/java/io/strimzi/operator/cluster/model/ClusterCa.java#L196

Reply:
_Please see the reply on the other comment_